Ingestion error when querying failed update from `:xt/txs`

```clojure
(deftest test-issue
  (xt/submit-tx tu/*node*
    [(xt/update-table :users '{:bind {:xt/id :john :age age}
                               :set {:age (inc age)}})])
  
  (t/is (thrown? clojure.lang.ExceptionInfo
          (xt/q tu/*node*
            '(from :xt/txs [xt/id])))))
```
Gives me the following exception:
```clojure
; ERROR in (test-issue) (await.clj:17)
; Uncaught exception, not in assertion.
; expected: nil
;   actual: clojure.lang.ExceptionInfo: Ingestion stopped: 
; {}
;  at xtdb.await$__GT_ingester_ex.invokeStatic (await.clj:17)
;     xtdb.await$__GT_ingester_ex.invoke (await.clj:16)
;     xtdb.await$notify_ex.invokeStatic (await.clj:58)
;     xtdb.await$notify_ex.invoke (await.clj:54)
;     xtdb.indexer.Indexer.indexTx (indexer.clj:659)
;     xtdb.log.watcher$watch_log_BANG_$reify__37507.acceptRecord (watcher.clj:56)
;     xtdb.log$tx_handler$fn__13383.invoke (log.clj:35)
;     clojure.core.protocols$iter_reduce.invokeStatic (protocols.clj:49)
;     clojure.core.protocols$fn__8246.invokeStatic (protocols.clj:75)
;     clojure.core.protocols/fn (protocols.clj:75)
;     clojure.core.protocols$fn__8194$G__8189__8207.invoke (protocols.clj:13)
;     clojure.core$reduce.invokeStatic (core.clj:6895)
;     clojure.core$reduce.invoke (core.clj:6877)
;     xtdb.log.NotifyingSubscriberHandler/fn (log.clj:91)
;     clojure.lang.AFn.run (AFn.java:22)
;     java.lang.Thread.run (Thread.java:833)
; Caused by: java.lang.UnsupportedOperationException: null
;  at xtdb.xtql.edn$parse_out_specs.invokeStatic (edn.clj:237)
;     xtdb.xtql.edn$parse_out_specs.invoke (edn.clj:225)
;     xtdb.xtql.edn$eval33393$fn__33395.invoke (edn.clj:625)
;     clojure.lang.MultiFn.invoke (MultiFn.java:229)
;     xtdb.indexer$$reify__34575.indexOp (indexer.clj:476)
;     xtdb.indexer.Indexer$fn__34595$index_tx_ops__34598.invoke (indexer.clj:606)
;     xtdb.indexer.Indexer$fn__34595$fn__34617.invoke (indexer.clj:618)
;     xtdb.indexer.Indexer/fn (indexer.clj:617)
;     xtdb.indexer.Indexer.indexTx (indexer.clj:581)
;     xtdb.log.watcher$watch_log_BANG_$reify__37507.acceptRecord (watcher.clj:56)
;     xtdb.log$tx_handler$fn__13383.invoke (log.clj:35)
;     clojure.core.protocols$iter_reduce.invokeStatic (protocols.clj:49)
;     clojure.core.protocols$fn__8246.invokeStatic (protocols.clj:75)
;     clojure.core.protocols/fn (protocols.clj:75)
;     clojure.core.protocols$fn__8194$G__8189__8207.invoke (protocols.clj:13)
;     clojure.core$reduce.invokeStatic (core.clj:6895)
;     clojure.core$reduce.invoke (core.clj:6877)
;     xtdb.log.NotifyingSubscriberHandler/fn (log.clj:91)
;     clojure.lang.AFn.run (AFn.java:22)
;     java.lang.Thread.run (Thread.java:833)
```

--- a/core/src/main/clojure/xtdb/xtql/edn.clj
+++ b/core/src/main/clojure/xtdb/xtql/edn.clj
@@ -620,6 +620,9 @@
             (when-not (map? set-specs)
               (throw (err/illegal-arg :xtql/malformed-set {:set set-specs, :update this})))
 
+            (when-not (vector? bind)
+              (throw (err/illegal-arg :xtql/malformed-bind {:bind bind, :update this})))
+
             (cond-> (DmlOps/update (str (symbol table)) (parse-col-specs set-specs this))
               for-valid-time (.forValidTime (parse-temporal-filter for-valid-time :for-valid-time this))
               bind (.binding (parse-out-specs bind this))

--- a/src/test/clojure/xtdb/api_test.clj
+++ b/src/test/clojure/xtdb/api_test.clj
@@ -535,6 +535,13 @@ VALUES (2, DATE '2022-01-01', DATE '2021-01-01')")])
           (t/is (= tx2-expected (all-users tx2)))
           (t/is (= tx1-expected (all-users tx1))))))))
 
+(t/deftest test-update-ingestion-error-3035
+  (xt/submit-tx tu/*node*
+    [(xt/update-table :users '{:bind {:xt/id :john :age age}
+                               :set {:age (inc age)}})])
+
+  (t/is (not (empty? (xt/q tu/*node* '(from :xt/txs [xt/id]))))))
+
 (t/deftest test-erase-xtql
   (letfn [(q [tx]
             (set (xt/q *node* '(from :foo [xt/id version xt/valid-from xt/valid-to])

--- a/src/test/clojure/xtdb/xtql/edn_test.clj
+++ b/src/test/clojure/xtdb/xtql/edn_test.clj
@@ -251,7 +251,12 @@
 
            (roundtrip-dml '(update :foo {:set {:version (inc v)},
                                          :bind [{:xt/id foo, :version v}]}
-                                   (from :bar [{:xt/id $bar-id, :foo foo}]))))))
+                                   (from :bar [{:xt/id $bar-id, :foo foo}])))))
+
+  (t/is (thrown-with-msg?
+          xtdb.IllegalArgumentException #"Illegal argument: ':xtql/malformed-bind'"
+          (roundtrip-dml '(update :foo {:bind {:not-a vector}
+                                        :set {:version (inc v)}})))))
 
 (t/deftest test-parse-delete
   (t/is (= '(delete :foo {:for-valid-time (in #inst "2020" nil),

